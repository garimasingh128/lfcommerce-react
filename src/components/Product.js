import React, { Component } from 'react';
import '../App.css';
import { Panel, Grid, Row, Col } from 'react-bootstrap';


class Product extends Component {
  constructor(props){
    super(props);
  }
  render() {
    return (
      <Panel className="no-border product-item">
          <div className="product-cover" style={{backgroundImage:`url(${this.props.productImage})`}}></div>
          <br />
          <Row>
              <Col md={8} xs={8}>
                <strong><a href={this.props.productURL}>{this.props.productName}</a></strong>
              </Col>
              <Col md={4} xs={4} className="text-right">
              {this.props.productPrice}
              </Col>
          </Row>
      </Panel>
    );
  }
}

export default Product;