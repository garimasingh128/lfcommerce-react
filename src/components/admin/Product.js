import React, { Component } from 'react';
import '../../App.css';
import { 
    Navbar, Nav, NavItem, NavDropdown,
    MenuItem, Badge, Grid, Row, Col,
    FormGroup, FormControl, Button } from 'react-bootstrap';
import {
  Link
} from 'react-router-dom';

class Product extends Component {
  render() {
    return  (
        <Grid style={{marginBottom:'30px'}}>
            <h3>Products</h3>
            <Breadcrumb>
                <Breadcrumb.Item href="/admin/dashboard">
                Dashboard
                </Breadcrumb.Item>
                <Breadcrumb.Item active>
                Products
                </Breadcrumb.Item>
            </Breadcrumb>
            <Row>
            <Col md={12}>
            <ListGroup>
                <ListGroupItem>
                    <Row>
                        <Col md={2}>
                            Nicholas Chen
                        </Col>
                        <Col md={8}>
                            nicholas_chan82@hotmail.com
                        </Col>
                        <Col md={2}>
                        Delete
                        </Col>
                    </Row>
                </ListGroupItem>
                <ListGroupItem>
                    <Row>
                        <Col md={2}>
                            Nicholas Chen
                        </Col>
                        <Col md={8}>
                            nicholas_chan82@hotmail.com
                        </Col>
                        <Col md={2}>
                        Delete
                        </Col>
                    </Row>
                </ListGroupItem>
            </ListGroup>
            </Col>
            </Row>
            <Button bsStyle="primary" style={{float:'right'}}>Checkout</Button>
        </Grid>)
  }
}

export default Product;