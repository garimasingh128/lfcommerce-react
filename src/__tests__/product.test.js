import React from 'react';
import {shallow, mount, render} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Product from './Product';

test('product component with display a product image, name and price', () => {
  const product = shallow(<Product productName="Test product"
                                  productURL="products/test-product-1"
                                  productImage="http://assets.teenvogue.com/photos/57dfbfe5046b3a2e2a7364bd/master/pass/Kylie-Armytee.jpg"
                                  productPrice='$12.90' />)
  const json = toJSON(product)
  expect(json).toMatchSnapshot()
});