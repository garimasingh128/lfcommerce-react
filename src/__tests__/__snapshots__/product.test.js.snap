// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product component with display a product image, name and price 1`] = `
<Panel
  bsClass="panel"
  bsStyle="default"
  className="no-border product-item"
  defaultExpanded={false}
>
  <div
    className="product-cover"
    style={
      Object {
        "backgroundImage": "url(http://assets.teenvogue.com/photos/57dfbfe5046b3a2e2a7364bd/master/pass/Kylie-Armytee.jpg)",
      }
    }
  />
  <br />
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={8}
      xs={8}
    >
      <strong>
        <Link
          replace={false}
          to="products/test-product-1"
        >
          Test product
        </Link>
      </strong>
    </Col>
    <Col
      bsClass="col"
      className="text-right"
      componentClass="div"
      md={4}
      xs={4}
    >
      $12.90
    </Col>
  </Row>
</Panel>
`;
